tests/battDB/test_views.py

```diff
@@ -647,3 +647,21 @@ def test_upload_view_unparsed_data(self):
         self.assertContains(
             get_response, "This file was uploaded without being processed."
         )
+
+    def test_invalid_form(self):
+        import os
+
+        from liionsden.settings import settings
+
+        # Login
+        self.client.post(
+            "/accounts/login/",
+            {"username": "test_contributor", "password": "contributorpass"},
+        )
```

Comment:
This isn't wrong, but I have typically used [the `client.login` method](https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.Client.login) for logging in in tests.